ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/RawStringInterningSpec.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class RawStringInterningSpec extends AnyFlatSpec with Matchers {
+
+  behavior of "RawStringInterning.from"
+
+  it should "start empty" in {
+    val current = RawStringInterning.from(Nil)
+    current.map shouldBe empty
+    current.idMap shouldBe empty
+    current.lastId shouldBe 0
+  }
+
+  it should "append empty entries to existing cache" in {
+    val previous = RawStringInterning(Map("one" -> 1), Map(1 -> "one"), 1)
+    val current = RawStringInterning.from(Nil, previous)
+    current.map shouldBe previous.map
+    current.idMap shouldBe previous.idMap
+    current.lastId shouldBe previous.lastId
+  }
+
+  it should "append non-empty entries to empty cache" in {
+    val current = RawStringInterning.from(List(1 -> "one"))
+    current.map shouldBe Map("one" -> 1)
+    current.idMap shouldBe Map(1 -> "one")
+    current.lastId shouldBe 1
+  }
+
+  it should "append non-empty entries to non-empty cache" in {
+    val previous = RawStringInterning(
+      Map("one" -> 1, "two" -> 2),
+      Map(1 -> "one", 2 -> "two"),
+      2,
+    )
+    val current = RawStringInterning.from(List(3 -> "three"), previous)
+    current.map shouldBe Map("one" -> 1, "two" -> 2, "three" -> 3)
+    current.idMap shouldBe Map(1 -> "one", 2 -> "two", 3 -> "three")
+    current.lastId shouldBe 3
+  }
+
+  behavior of "RawStringInterning.newEntries"
```

Comment:
can you please add one more bigger involved case (like having 3 elements already, adding 5 which from 3 are redundant, and observing new items with correctly assigned id-s)